src/features/auth/api.ts

```diff
@@ -2,18 +2,31 @@ import { api } from 'api/api'
 import { AccountRequest, SigninRequest } from 'api/gen'
 import { useLoginRoutine } from 'features/auth/AuthContext'
 
-export function useSignIn(): (data: SigninRequest) => Promise<boolean> {
+type SignInResponse =
+  | {
+      status: true
```

Comment:
Tu préfères pas mettre un number comme indiqué dans le type ? Ca me fait bizarre de voir status associé à `true` ou `false`, contrairement à `1` ou `0` :
```suggestion
      status: 1
```

Reply:
l'utilisation de 1 et 0 est moins représentative du succès/échec que le booléen quand on a que 2 valeurs possibles.

sinon, autant utiliser une énumération plus explicite.

ou alors renommer status en quelque chose de plus explicite dans le contexte: `isSuccess` par exemple ?

t'en penses quoi ?

Reply:
l'autre avantage d'utiliser un booléen c'est de faciliter l'usage:
```ts
if (response.isSuccess)
```